converter.py

```diff
@@ -0,0 +1,275 @@
+                    # newer version of notebooks includes a
+                    # Table of Contents automatically in the first
+                    # cell, skip that and find the next level 1 header
+                    if title != 'Table of Contents':
+                        break
+        return title
+
+    def _construct_header(self):
+        """Create a knowledge repo style header as a dictionary."""
+
+        def flatten_list(l):
+            """
+            Although not needed for the current version, we could
+            have multiple tags and authors, in that case we would
+            need to flatten them out.
+            """
+            flat = []
+            for item in l:
+                if isinstance(item, list):
+                    flat += item
+                else:
+                    flat.append(item)
+
+            return flat
+
+        header = {'cell_type': 'raw', 'metadata': {}}
+
+        # header text required by the knowledge repo
+        # 1. a '- ' in front is required for knowledge repo tag
+        # 2. tldr is a required field, fill it with notebook's
+        # original github link for now
+        header_text = [
+            '---',
+            'title: {}'.format(self.title_),
+            'authors:',
+            '- {}'.format(self.AUTHOR),
+            'tags:',
+            '- ' + self.tags_,
+            'created_at: {}'.format(self.date_created_),
+            'updated_at: {}'.format(self.date_updated_),
+            'tldr: Nothing for tldr section as of now.',
+            '---']
+
+        header_text = flatten_list(header_text)
+        header_text = [text + '\n' for text in header_text[:-1]] + [header_text[-1]]
+        header['source'] = header_text
+        return header
+
+    def _construct_github_link_cell(self):
+        """Add a cell that contains link to original notebook on github"""
+        github_link_cell = {
+            'cell_type': 'markdown',
+            'metadata': {},
+            'source': ['Link to original notebook: {}'.format(self.github_link_)]}
+        return github_link_cell
+
+    def add(self, notebook):
+        """
+        Add the converted notebook to the knowledge repo.
+
+        Parameters
+        ----------
+        notebook : dict
+            Jupyter notebook's raw json represented in dictionary format.
+        """
+        with open(self._path, 'w', encoding = 'utf-8') as f:
+            json.dump(notebook, f)
+
+        # create a run knowledge repo command
+        destination = os.path.join(self.knowledge_repo_path, 'project', self.tags_)
+        cmd = 'knowledge_repo --repo {} add {} -p {}'.format(
+            self.knowledge_repo_path, self._path, destination)
+
+        p = subprocess.Popen(
+            cmd, stdout = subprocess.PIPE, stdin = subprocess.PIPE,
+            stderr = subprocess.STDOUT, shell = True)
+        p.communicate(input = b'generated by automated airbnb knowledge repo setup')
+
+        if not self.inplace:
```

Comment:
It can be helpful to inspect the converted notebooks, so I'd add a flag for this option

Reply:
I believe the knowledge repo also stores a copy.